diff --git a/common/cpp/react/renderer/components/safeareacontext/RNCSafeAreaViewShadowNode.cpp b/common/cpp/react/renderer/components/safeareacontext/RNCSafeAreaViewShadowNode.cpp
--- a/common/cpp/react/renderer/components/safeareacontext/RNCSafeAreaViewShadowNode.cpp
+++ b/common/cpp/react/renderer/components/safeareacontext/RNCSafeAreaViewShadowNode.cpp
@@ -16,9 +16,9 @@
     Style::Length edge,
     Style::Length axis,
     Style::Length defaultValue) {
-  if (edge.unit() != Unit::Undefined) {
+  if (!edge.isUndefined()) {
     return edge;
   }
-  if (axis.unit() != Unit::Undefined) {
+  if (!axis.isUndefined()) {
     return axis;
   }
   return defaultValue;
